community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/DefaultNodeCursor.java

```diff
@@ -242,21 +246,27 @@ public boolean shouldRetry()
     @Override
     public void close()
     {
-        if ( pageCursor != null )
-        {
-            pageCursor.close();
-            pageCursor = null;
-        }
         read = null;
+        hasChanges = HasChanges.MAYBE;
+        addedNodes = emptySet();
+        reset();
 
         if ( labelCursor != null )
         {
             labelCursor.close();
             labelCursor = null;
         }
-        hasChanges = HasChanges.MAYBE;
-        addedNodes = emptySet();
-        reset();
+
+        if ( pageCursor != null )
+        {
+            pageCursor.close();
+            pageCursor = null;
+
+            if ( pool != null )
```

Comment:
Why are we doing this null check? Are we expecting this pool to be null in some real scenarios? If this check is here to allow tests to pass in null then I think that they should pass in a no-op consumer instead so that we can get rid of this condition. (many similar places below too).